packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1256,7 +1325,14 @@ export class TransactionController extends BaseControllerV1<
       blockTracker: networkClient.blockTracker,
       getCurrentAccount: this.getSelectedAddress,
       getLastFetchedBlockNumbers: () => this.state.lastFetchedBlockNumbers,
-      getNetworkState: this.getNetworkState, // TODO: fake this via networkClient
+      getNetworkState: () => {
+        return {
+          selectedNetworkClientId: 'mainnet',
+          networkConfigurations: {},
+          networksMetadata: {},
+          providerConfig: {} as ProviderConfig,
+        };
```

Comment:
Does this work? Looks like we need this to be: 
```suggestion
      getNetworkState: () => {
        return {
          providerConfig: { chainId } as ProviderConfig,
        };
```

since it only needs the chainId off the (faked) providerConfig but your not passing it?

Reply:
reason for this is i just didnt want it to call `this.getNetworkState` because that gets mocked differently in tests and it gets called in the constructor now